### ISSUE TITLE
Drag Handle

### ISSUE THREAD

**Issue by jsthivierge** [tiberiuzuld/angular-gridster2#66]
Hi,

I would like to know if it is possible to specify multiple ignoreContentClasses for the draggable feature. Or if it is possible to specify instead a draggable area class.

My use case is that I have a box with the box header separated in a title area (draggable) and an action buttons area (not draggable) and then the box content (not draggable). I would like my box to be draggable only by the title area (within the header).

Thanks

**Comment by tiberiuzuld** [comment]
Hi @jsthivierge ,

Try this:
```html
<gridster-item>
  <div class="gridster-item-content">
      Some content to select and click without dragging the widget
  </div>
  <div class="toolbar">
     <div class="drag-handle"></div>
     <div class="item-buttons gridster-item-content">
           <button md-icon-button md-raised-button class="drag-handler">
              <md-icon>open_with</md-icon>
          </button>
          <button md-icon-button md-raised-button class="remove-button" (click)="removeItem($event, item)"
            (touchstart)="removeItem($event, item)" mdTooltip="Remove">
            <md-icon>clear</md-icon>
          </button>
     </div>
  </div>
</gridster-item>
```

Let me know if this works for you.

**Comment by jsthivierge** [comment]
Hi @tiberiuzuld,

It works to copy the class gridster-item-content to all the places we do not want the drag to occur, but would it make more sense in the future if we only had to specify the class once for where it is draggable instead of the other way around ? (See below)

Here's my pseudo-markup for what I'm trying to achieve:

```html
<box>
    <header>
          <div class="draggable-area">Title</div>
          <div class="action-buttons">...</div>
    </header>
    <content class="gridster-item-content">
          ...
    </content>
</box>
```

The fact is that I'm switching from angular-gridster and I had only to specify a class for the draggable handle (in my case .draggable-area) then. Therefore I could for example apply additional css styles when the area is draggable and I could activate dynamically some areas to be draggable or not.

Thanks a lot,
JS

**Comment by tiberiuzuld** [comment]
Hi @jsthivierge ,

Currently on `dragStart` there is a check for the class `gridster-item-content` if is present on the event propagation through the DOM. Code [here](https://github.com/tiberiuzuld/angular-gridster2/blob/master/src/lib/gridsterDraggable.service.ts) line 59.

Reversing this will make a breaking change and on initialization of each widget the event for `dragStart` will have to add the event on the child element(s) with class `draggable-area`.
If the user doesn't add any class on his child elements will not have drag on his widgets. 

What if the child element(s) are loaded with lazy loading ? 

I think this change will create to much headache.

Thanks,

**Comment by jsthivierge** [comment]
I'm not sure that I understand why it would be troublesome to check on `dragStart` if the child element is a descendant of an element with class `draggable-area`.

You're right, it would be a breaking change if reversing this. Could supporting both be a solution ? (`ignoreContentClass` and `dragHandle`) If there are no conflicts between them.

If you say it is too troublesome, let it be. It's only a suggestion :-)

Thanks

**Comment by tiberiuzuld** [comment]
Hi @jsthivierge ,

Hmm yes you are right I could do the same and check for `dragHandleClass` at the same time I check for `ignoreContentClass`.

I don't know why I was thinking to add the `dragStart` event on that element.

Yes the new option would be easy to add. 
But to be able not to add a breaking change will need to add another option.
Something like `ignoreContent` by default `false`. 
If `true` then will ignore any event from inside widget unless is from `dragHandleClass`.

Will see when I have time to implement this. Maybe will be able to squeeze it this week since it will not take to much.

Thanks,

**Comment by jsthivierge** [comment]
Sounds good!

Cool thanks a lot
JS

**Comment by tiberiuzuld** [comment]
Hi @jsthivierge ,
Implemented in v3.2.0.

Let me know if you have any more issues.

Thanks,
